### PR TITLE
Add site metadata to each chart module.

### PR DESCRIPTION
**Pull request by samussiah**
TODO: figure out why bar charts no longer display thresholds

## Overview
This PR adds a parameter named `dfSite` that optionally passes site metadata through to the scatter plot, bar chart, and time series widgets.

## Test Notes/Sample Code
```
snapshot <- Make_Snapshot(
    lPrevSnapshot = readRDS(system.file(
        'data-longitudinal',
        'longitudinal_snapshot.rds',
        package = 'clindata'
    ))
)
snapshot$lCharts$kri0001$scatterJS
snapshot$lCharts$kri0001$barScoreJS
snapshot$lCharts$kri0001$barMetricJS
snapshot$lCharts$kri0001$timeSeriesContinuousScoreJS
```

Notes: 
Verify each chart type displays site metadata on hover (investigator name / # of participants).

### CODE REVIEW

File: R/Widget_TimeSeries.R [hunk 1]
```diff
@@ -97,11 +98,16 @@ Widget_TimeSeries <- function(
...
+  if (!is.null(dfSite)) {
+    dfSite <- jsonlite::toJSON(dfSite, na = "string")
+  }
+
   # forward options using x
   x <- list(
     dfSummary = jsonlite::toJSON(dfSummary, na = "string"),
     lLabels = lLabels,
     dfParams = dfParams,
+    dfSite = dfSite,
```
Comment (jwildfire) [inline R/Widget_TimeSeries.R:110]: @samussiah Assuming new version of rbmviz knows how to handle NULL for dfSite?

**Review by jwildfire — APPROVED** [review]
Works well for me with given example code (with and without stacked snapshots). 

A couple of minor questions from code review for you and then I think we're good to merge.

**Comment by jwildfire** [comment]
Fixes #1522

File: R/Widget_TimeSeries.R (hunk 1 above)
Comment (samussiah) [reply R/Widget_TimeSeries.R:110]: @jwildfire - yep, completeely optional.

**Pull request merged**